### PR TITLE
Add unit test for `calc_field` method

### PR DESCRIPTION
**Pull request by danielhollas**
Extracted from #26

### CODE REVIEW

File: src/promdens/__init__.py [hunk 1]
```diff
@@ -1 +1 @@
-from .promdens import main
```
Comment (danielhollas) [inline src/promdens/__init__.py:1]: I think it's time to export these classes before we publish to PyPI.

File: tests/test_pulse.py [hunk 2]
```diff
@@ -363,4 +363,4 @@ def test_pulse_wigner_de_equals_omega(make_pulse, envelope_type):
     for i, tprime in enumerate(tprimes):
         effective_omega = omega + 2 * chirp * tprime
         de = effective_omega
-        assert pulse.wigner_transform(tprime, de) == pytest.approx(s[envelope_type][i], abs=1e-15, rel=1e-15)
```
Comment (danielhollas) [inline tests/test_pulse.py:366]: I think unfortunately putting the snapshots inside `pytest.approx` doesn't quite work and these tests were failing for me locally for some reasons, even though they were passing these tolerances.

Comment (JanosJiri) [reply tests/test_pulse.py:366]: Interesting, I had problems with them too. It never showed me the tolerance when I used the value from the snapshot. I haven't thought about using `approx` on the other value.

**Review by JanosJiri — APPROVED** [review]
Awesome, thanks for that. I haven't plotted the values but I believe they are correct since we plot the envelopes with `envelopes_analysis.py`. It seems now we have all the major parts of the code tested.

**Pull request merged**